packages/jupyter-ai/jupyter_ai/chat_handlers/export.py

```diff
@@ -25,21 +26,23 @@ def chat_message_to_markdown(self, message):
         else:
             return ""
 
-    # Multiple chat histories in separate files
-    def get_chat_filename(self, path="./chat_history.md"):
-        filename, extension = os.path.splitext(path)
-        counter = 1
-        while os.path.exists(path):
-            path = filename + "_" + str(counter) + ".md"
-            counter += 1
+    # Multiple chat histories can be saved in separate files, each with a timestamp
+    def get_chat_filename(self, file_id):
+        file_id = file_id.split(" ")
+        if len(file_id) > 1:
+            file_id = file_id[-1]
+        else:
+            file_id = "chat_history"
+        outfile = f"{file_id}-{datetime.now():%Y-%m-%d-%H-%M}.md"
+        path = os.path.join(self.root_dir, outfile)
         return path
 
     async def process_message(self, _):
```

Comment:
```suggestion
    async def process_message(self, message: HumanChatMessage):
```

Reply:
The new option for line 40 is now used. This facilitates parsing the output file name from the message passed in.